src/common.rs

```diff
@@ -72,17 +89,17 @@ fn make_socket(host: &str, cid: u32) -> Result<(RawFd, Domain, SockAddr)> {
 
     match domain {
         Domain::Unix => {
-            fd = socket(
-                AddressFamily::Unix,
-                SockType::Stream,
-                SockFlag::SOCK_CLOEXEC,
-                None,
-            )
-            .map_err(|e| Error::Socket(e.to_string()))?;
-            let sockaddr_u =
-                UnixAddr::new_abstract(hostv[1].as_bytes()).map_err(err_to_others_err!(e, ""))?;
-            sockaddr = SockAddr::Unix(sockaddr_u);
+            fd = socket(AddressFamily::Unix, SockType::Stream, SOCK_CLOEXEC, None)
+                .map_err(|e| Error::Socket(e.to_string()))?;
+
+            // MacOS doesn't support atomic creation of a socket descriptor with SOCK_CLOEXEC flag,
+            // so there is a chance of leak if fork + exec happens in between of these calls.
+            #[cfg(target_os = "macos")]
+            set_fd_close_exec(fd)?;
+
+            sockaddr = SockAddr::Unix(make_addr(hostv[1])?);
         }
+        #[cfg(target_os = "linux")]
```

Comment:
Left the pattern Domain::Vsock for macos is not harmful

Reply:
Vsock is linux specific and it explicitly excluded in nix crate with cfg attributes, so it won't compile this way and we can't get rid of cfg attributes entirely here. So I think we should follow same patterns and exclude it on platforms that doesn't support it. This also will prevent users from using features that are not supported at compile time.

Reply:
ok, got it.